examples/custom_logger.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const Pulsar = require('../index.js');
+
+(async () => {
+    // Create a client
+    const client = new Pulsar.Client({
+        serviceUrl: 'pulsar://localhost:6650',
+        operationTimeoutSeconds: 30,
+    });
+
+    Pulsar.Client.setLogHandler((level, file, line, message) => {
+        console.log('[%s][%s:%d] %s', Pulsar.LogLevel.toString(level), file, line, message);
```

Comment:
```
[[object Object]][ClientConnection:189] [<none> -> pulsar://localhost:6650] Create ClientConnection, timeout=10000
[[object Object]][ConnectionPool:97] Created connection for pulsar://localhost:6650
[[object Object]][ClientConnection:379] [127.0.0.1:61358 -> 127.0.0.1:6650] Connected to broker
[[object Object]][HandlerBase:72] [persistent://public/default/my-topic, ] Getting connection from pool
[[object Object]][ProducerImpl:190] [persistent://public/default/my-topic, ] Created producer on broker [127.0.0.1:61358 -> 127.0.0.1:6650] 

```
`Pulsar.LogLevel.toString(level)`  Should be converted to a string, as DEBUG, INFO, WARN, etc.

Reply:
Another question: How do users set the log level?

Reply:
See here: https://github.com/apache/pulsar-client-node/pull/253#discussion_r1038074426
This PR already did that.
The user can filter the log level but themself.

Reply:
Oh, sorry. I missed it. 

> The user can filter the log level by themself.

It looks like the default CPP is the log of the printed info level. If the user wants to see the debug log, where does he need to set it up?

Reply:
> It looks like the default CPP is the log of the printed info level. If the user wants to see the debug log, where does he need to set it up?

Currently, the log level is hardcoded [here](https://github.com/apache/pulsar-client-cpp/blob/7a7b3aa5faa802fa354d2b2e41d38ce22241d700/lib/c/c_ClientConfiguration.cc#L81). So we can not get the debug log in the nodejs client. But I think it's worth supporting getting debug log from cpp client.

Reply:
I have created an issue to track it: https://github.com/apache/pulsar-client-cpp/issues/137 PTAL.